modules/core/src/main/scala/lucuma/odb/api/model/ScienceRequirements.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats.Eq
+import cats.syntax.all._
+import cats.data.State
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import lucuma.core.optics.syntax.lens._
+import lucuma.odb.api.model.syntax.input._
+import clue.data.Input
+import monocle.Lens
+import monocle.Focus
+
+final case class ScienceRequirements(
+  mode:                     ScienceMode,
+  spectroscopyRequirements: SpectroscopyScienceRequirements
```

Comment:
Nothing would prevent combining an `Imaging` mode with this?  One idea would be to make `ScienceRequirements` a `sealed trait` with a `mode` field and have `SpectroscopyRequirements` be an instance?   Or perhaps you want to store both imaging and spectroscopy requirements, if the user should enter both, and then use the `mode` to determine which one they really want?

Reply:
I want to store both. This is an UI factor as I want to store the parameters in either case in case you switch.
Let's say you're in spectroscopy and change some params, then look at imaging and then return
Ideally we'd preserve the previously changed params.

Reply:
That makes sense.  :+1: